src/Config.php

```diff
@@ -40,11 +40,13 @@ public function __construct($name = 'default')
     }
 
     /**
+     * @param string $name
+     *
      * @return static
      */
-    public static function create()
+    public static function create($name = 'default')
```

Comment:
Please add a test showing that this new argument is forwarded to the constructor.

Reply:
Thought about it. Is it *really* necessary for that? I thought it'd add more useless tests than anything else. :(

Reply:
Actually I'm not even sure this change is desirable, it looks dangerous to me. If a subclass overrides the constructor with a different argument than `$name` (so likely with a different meaning), it will receive an unexpected value.

@SpacePossum @kubawerlos @keradus What's your opinion?

Reply:
That was the thought process:

- Stumbled upon the problem that the documentation provides a way to create a config.
- Couldn't name it other than "default" that way.
- Tried instanciating with new(), which works but is not the intended way.
- Added the option to name the config and propagate.
- Thought that this could also be solved with a setter (expected, but not found).
- Added a setter.

Now:

- Removed setter.
- Solved setter failing test.
- Added test for the static constructor with a name and without a name.
- Removed the setter prototype from the Interface.

If you think that's not a problem from start, fine. But I still stumbled upon it. :)

Reply:
I understand the process that led to your changes proposal, it souds legit, but I feel it is kind of a BC break. IMO the real issue is that `Config::create()` uses `new static()`. It should use `new self()` so that we know for sure which constructor signature is used.

Anyway, since the constructor is public, the `Config::create()` has not much value so maybe we can deprecate it instead of trying to improve it?

Reply:
:+1:  for deprecating `Config::create()` on master and remove on `3.0`, as I see no value over it as one can just use the constructor, less is more :)

Reply:
And let's not forget to update the README :)

Seems like a fine update!

Reply:
Good point! that we can already do on 2.15 and merge upstream to encourage people to use the constructor :)

Reply:
Shall I close this PR so that we focus on a new one or work here and update the name?

Reply:
I'm fine with reusing this one, than other reviewers can catch up how the PR came to be :)

Reply:
So, to sum up:
- I'll remove everything I've done, including tests.
- Replace the README constructor 
- Add a @deprecated PHPDoc block to the static constructor?

Reply:
yeah looking good already, I do have a request I just thought about though...
could you also update this call;
https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/master/.php_cs.dist#L13

this way we won't be calling the deprecated method ourselves anymore :)

Reply:
Oh, crap. The tests are a nightmare, this construction was used everywhere.
I don't have so much time right now, I'm gonna have to update them all, there are tenths of new tests failing that I have to update right now. 🗡